docs/data/data-grid/performance/performance.md

```diff
@@ -0,0 +1,47 @@
+# Data Grid - Performance
+
+<p class="description">Improve the performance of the DataGrid using the recommendations from this guide.</p>
+
+## Memoize inner components with `React.memo`
+
+The `DataGrid` component is composed of a central state object where all data is stored.
+When an API method is called, a prop changes, or the user interacts with the UI (e.g. filtering a column), this state object is updated with the changes made.
+To reflect the changes in the interface, the component must re-render.
+Since the state behaves like `React.useState`, it means that the `DataGrid` component will re-render as well as its children, which includes column headers, rows and cells.
```

Comment:
```suggestion
Since the state behaves like `React.useState`, the `DataGrid` component will re-render its children, including column headers, rows, and cells.
```
To keep it shorter